module 05 initial commit

mwand opened the pull request:
Rewrote the slides to stick to TDD methodology.  This turned out to lead to a much smoother flow!

abhutta0 commented:
already have pattern 1 and 2 in observer ...
then next one is numbered as pattern #2. Might be confusing to students. May wants to change the two in observer to version 1 and version 2.

mwand commented:
Added "data-pull" as a pattern and renumbered patterns as 1 - 5.

Slide #34: the issue is that you are constrained not to add any more methods.  If one of your server methods is already a handler, it's ok to pass this.theHandlerMethod .  I don't think this is a big deal, so the new slide doesn't emphasize it-- it just does it.

Added some transition in the speaker's notes for Singleton.

I updated _data/module.yml to get nicer module names, but now that's in conflict w with main.  Please resolve these with the module names you like.

The activity is unchanged.

abhutta0 commented:
@mwand resolved conflict. Are we going to add a word for "structural scale"?

rest lgtm. It is ready to merge.
